openpype/hooks/pre_ocio_hook.py

```diff
@@ -0,0 +1,60 @@
+from openpype.lib import PreLaunchHook
+
+from openpype.pipeline.colorspace import (
+    get_imageio_config,
+    is_host_use_ocio_config_activated
+)
+from openpype.pipeline.template_data import get_template_data_with_names
+
+
+class OCIOEnvHook(PreLaunchHook):
+    """Set OCIO environment variable for hosts that use OpenColorIO."""
+
+    order = 0
+    app_groups = [
```

Comment:
would it be possible to expose these on the settings so if we leave it empty it defaults to all apps?

Reply:
Hey @fabiaserra thanks for the feedback. Yeah, I was having the same idea, but then I was also thinking why would anybody touch those settings - since it is possible either disable the color management for the host or disable `use_ocio_config` attribute in the host's image.

Reply:
But anybody can always add OCIO env-var into global or application environments in settings. This would be unnecessary configuration. Why would anybody want to add OCIO ever to host and disable host color management? Every new host integration should also add host name into the hook as hardcoded rather than adding it via the settings.

Reply:
> Every new host integration should also add host name into the hook as hardcoded rather than adding it via the settings.
With this you are expecting that every user needs to fork OP to change this file and add their custom flavor as opposed to allowing the customization on the settings level (not needing to touch any code)

Reply:
> With this you are expecting that every user needs to fork OP to change this file and add their custom flavor as opposed to allowing the customization on the settings level (not needing to touch any code)

Thanks @fabiaserra for your point. Filtering by `app_groups` was not the optimal design, it would need to be done by forking in case you would need to add a new **app group** via the system settings. I have changed the filtering to be per hosts. This way if any new host integration is added, it would be always done via code changes and hardcoding it makes more sense.

Reply:
Fair enough, then we certainly need some color management documentation that explains this clearly and how would you go on supporting it on new hosts.

Apologies as this is a lazy question that I could probably test myself but have you checked how having the $OCIO defined plays out with the `use_ocio_config` and `activate_host_color_management` from this PR?

Reply:
> but have you checked how having the $OCIO defined plays out with the use_ocio_config and activate_host_color_management from this PR?

Did and it worked as expected. Please test it as that would be also very valuable input.

Reply:
Tested it in Maya and unfortunately, with `activate_host_color_management` (we could call it `enable_color_management_host` so it's more aligned with the label or viceversa?) I'm seeing the default Maya OCIO enabled, shouldn't that be disabled then?:
![image](https://user-images.githubusercontent.com/4348536/233611649-26ae3ac1-1770-4f3f-b53c-3a1e0c1957e2.png)

However, for the workflow that I'm interested in, this does seem to work fine:
![image](https://user-images.githubusercontent.com/4348536/233612955-87f1e266-904a-4d50-9793-90e7f75b7d48.png)

Reply:
What's the recommended workflow for cleaning up deprecated settings from older versions to avoid getting warnings like:
```
*** WRN: >>> { DictImmutableKeysEntity }: [  project_settings/fusion/imageio Unknown keys in default values: "ocio"  ] 
*** WRN: >>> { DictImmutableKeysEntity }: [  project_settings/global/imageio/file_rules Unknown keys in studio overrides: "enabled"  ] 
*** WRN: >>> { DictImmutableKeysEntity }: [  project_settings/fusion/imageio Unknown keys in default values: "ocio"  ] 
*** WRN: >>> { DictImmutableKeysEntity }: [  project_settings/global/imageio/file_rules Unknown keys in studio overrides: "enabled"  ] 
*** WRN: >>> { DictImmutableKeysEntity }: [  project_settings/global/imageio/file_rules Unknown keys in studio overrides: "enabled"  ]
```